app/components/forms/session-speaker-form.js

```diff
@@ -302,6 +302,16 @@ export default Component.extend(FormMixin, {
     return groupBy(this.get('fields').toArray(), field => field.get('form'));
   }),
 
+  // Clicking on the add session button creates a blank record which increases the length of the speaker's list by 1.
+  noSpeakerExists: computed('speakers', function() {
```

Comment:
@pradeepgangwar this is not the route. The route is public and we don't create a speaker when the session is created in the public route as the speaker is the logged in user.

Reply:
We can change this later. We need to change a lot of logic in session-speaker-form.js to do this my way. But since this will work for now we can go for this. 
Although that 1 record comes because in above code snippet when e are creating the new record for speaker we also set its relationship with event. And hence it also gets added to our query list here. Hence we always get that record with event although its empty.
https://github.com/fossasia/open-event-frontend/blob/0edc77cabbd31f49cbbb746a1a1f30aeb724d5fd/app/routes/events/view/sessions/create.js#L15-L16